### ISSUE TITLE
Name.pl Missing from /lib/unicore/

### ISSUE THREAD

**Issue by p5pRT** [Perl/perl5#16053]
Migrated from [rt.perl.org#131669](https://rt-archive.perl.org/perl5/Ticket/Display.html?id=131669) (status was 'rejected')

Searchable as RT131669$

**Comment by p5pRT** [comment]
### From paul.lemmons@tmcaz.com

#### Created by paul.lemmons@tmcaz.com



When trying to install a module from cpan \(Text&#8203;::CSV\) the error message

&nbsp;   Unknown charname 'EURO SIGN' at t/51\_utf8\.t line 169\, within string

is encountered\. It turnes out that this is caused by the file
Name\.pl being missing from the lib/unicore directory\. Problem is 
resolved locally by the following steps&#8203;:

wget http&#8203;://www\.cpan\.org/src/5\.0/perl\-5\.20\.1\.tar\.gz
gunzip \-c \.\./perl\-5\.20\.1\.tar\.gz | tar xvf \-
cd perl\-5\.20\.1/lib/unicore/
chmod 755 \./mktables
\./mktables \# Note there is no output\.\.\. just wait\. It will complete
sudo cp Name\.pl /usr/opt/perl5/lib/5\.20\.1/unicore

The simple solution to this bug is to have the file packaged along with 
Perl on AIX 7\.2\.

<details><summary>Perl Info</summary>

```

Flags:
     category=core
     severity=medium

Site configuration information for perl 5.20.1:

Configured by root at Fri Feb  6 14:40:27 CST 2015.

Summary of my perl5 (revision 5 version 20 subversion 1) configuration:
      Platform:
     osname=aix, osvers=6.1.0.0, archname=aix-thread-multi
     uname='aix blade08 1 6 00003c3ad100 '
     config_args='-d -Dprefix=/usr/opt/perl5 -Dcc=xlc_r -Duseshrplib 
-Dusethreads'
     hint=recommended, useposix=true, d_sigaction=define
     useithreads=define, usemultiplicity=define
     use64bitint=undef, use64bitall=undef, uselongdouble=undef
     usemymalloc=n, bincompat5005=undef
   Compiler:
     cc='xlc_r -q32', ccflags ='-D_ALL_SOURCE -D_ANSI_C_SOURCE 
-D_POSIX_SOURCE -qmaxmem=-1 -qnoansialias -qlanglvl=extc99 
-DUSE_NATIVE_DLOPEN -DNEED_PTHREAD_INIT -q32 -D_LARGE_FILES',
     optimize='-O',
     cppflags='-D_ALL_SOURCE -D_ANSI_C_SOURCE -D_POSIX_SOURCE 
-qmaxmem=-1 -qnoansialias -qlanglvl=extc99 -DUSE_NATIVE_DLOPEN 
-DNEED_PTHREAD_INIT'
     ccversion='12.1.0.9', gccversion='', gccosandvers=''
     intsize=4, longsize=4, ptrsize=4, doublesize=8, byteorder=4321
     d_longlong=define, longlongsize=8, d_longdbl=define, longdblsize=8
     ivtype='long', ivsize=4, nvtype='double', nvsize=8, Off_t='off_t', 
lseeksize=8
     alignbytes=8, prototype=define
   Linker and Libraries:
     ld='ld', ldflags =' -brtl -bdynamic -b32'
     libpth=/lib /usr/lib /usr/ccs/lib
     libs=-lbind -lnsl -ldbm -ldl -lld -lm -lcrypt -lpthreads -lc
     perllibs=-lbind -lnsl -ldl -lld -lm -lcrypt -lpthreads -lc
     libc=, so=a, useshrplib=true, libperl=libperl.a
     gnulibc_version=''
   Dynamic Linking:
     dlsrc=dl_aix.xs, dlext=so, d_dlsymun=undef, ccdlflags=' 
-bE:/usr/opt/perl5/lib/5.20.1/aix-thread-multi/CORE/perl.exp'
     cccdlflags=' ', lddlflags='-bhalt:4 -G -bI:$(PERL_INC)/perl.exp 
-bE:$(BASEEXT).exp -bnoentry -lpthreads -lc -lm'



@INC for perl 5.20.1:
     /usr/opt/perl5/lib/site_perl/5.20.1/aix-thread-multi
     /usr/opt/perl5/lib/site_perl/5.20.1
     /usr/opt/perl5/lib/5.20.1/aix-thread-multi
     /usr/opt/perl5/lib/5.20.1
     /usr/opt/perl5/lib/site_perl
     .


Environment for perl 5.20.1:
     HOME=/home/tspdlp
     LANG=C
     LANGUAGE (unset)
     LC__FASTMSG=true
     LD_LIBRARY_PATH (unset)
     LIBPATH (unset)
     LOGDIR (unset)
 
PATH=/usr/vacpp/bin:/usr/vatools/bin:/usr/bin:/etc:/usr/sbin:/usr/ucb:/home/tspdlp/bin:/usr/bin/X11:/sbin:/usr/dt/bin:/usr/local/bin:/epic/bin:.
     PERL_BADLANG (unset)
     SHELL=/usr/bin/bash

```
</details>

**Comment by p5pRT** [comment]
### From @iabyn

On Tue\, Jun 27\, 2017 at 02&#8203;:39&#8203;:51PM \-0700\, Paul Lemmons wrote&#8203;:
> When trying to install a module from cpan \(Text&#8203;::CSV\) the error message
> 
>    Unknown charname 'EURO SIGN' at t/51\_utf8\.t line 169\, within string
> 
> is encountered\. It turnes out that this is caused by the file
> Name\.pl being missing from the lib/unicore directory\. Problem is 
> resolved locally by the following steps&#8203;:
> 
> wget http&#8203;://www\.cpan\.org/src/5\.0/perl\-5\.20\.1\.tar\.gz
> gunzip \-c \.\./perl\-5\.20\.1\.tar\.gz | tar xvf \-
> cd perl\-5\.20\.1/lib/unicore/
> chmod 755 \./mktables
> \./mktables \# Note there is no output\.\.\. just wait\. It will complete
> sudo cp Name\.pl /usr/opt/perl5/lib/5\.20\.1/unicore
> 
> The simple solution to this bug is to have the file packaged along with 
> Perl on AIX 7\.2\.

A normal perl build and install will generate and install name\.pl in the
correct place\.

Your bug report doesn't make it clear whether you personally built and
installed this perl from the tarball and something went wrong\, or whether
you are using a pre\-packaged perl install and the file is missing from
that packaging\.

If the latter\, then you should take the issue up with whoever created the
package\.

If the former\, then you'll need to reproduce the issue with a perl
distribution which is still maintained; preferably 5\.26\.0\.

If you can show a run of  perl\-5\.26\.0\.tar\.gz being unpacked\, \./Configure
being run with \-Dprefix='/some/dir' where /some/dir is initially empty\,
and where name\.pl fails to be installed\, then we will look at the issue
further\.


\-\- 
More than any other time in history\, mankind faces a crossroads\. One path
leads to despair and utter hopelessness\. The other\, to total extinction\.
Let us pray we have the wisdom to choose correctly\.
&nbsp;    \-\- Woody Allen

**Comment by p5pRT** [comment]
The RT System itself - Status changed from 'new' to 'open'

**Comment by p5pRT** [comment]
### From paul.lemmons@tmcaz.com

Thank you\. It was packaged by IBM for AIX 7\.2\. So it is "the latter"\. I 
appreciate the quick response\.

\-\-\-\-\-\-\-\- Original Message \-\-\-\-\-\-\-\-
Subject&#8203;: Re&#8203;: \[perl Dave Mitchell via RT31669\] Name\.pl Missing from 
/lib/unicore/
From&#8203;: Dave Mitchell via RT \<perlbug\-followup@&#8203;perl\.org>
To&#8203;: \<paul\.lemmons@&#8203;tmcaz\.com>
Date&#8203;: 06/29/2017 02&#8203;:11 AM
> \*\*\*This message came from an external source\. Please exercise caution when opening any attachments or clicking on links\.\*\*\*
>
> On Tue\, Jun 27\, 2017 at 02&#8203;:39&#8203;:51PM \-0700\, Paul Lemmons wrote&#8203;:
>> When trying to install a module from cpan \(Text&#8203;::CSV\) the error message
>>
>>     Unknown charname 'EURO SIGN' at t/51\_utf8\.t line 169\, within string
>>
>> is encountered\. It turnes out that this is caused by the file
>> Name\.pl being missing from the lib/unicore directory\. Problem is
>> resolved locally by the following steps&#8203;:
>>
>> wget http&#8203;://www\.cpan\.org/src/5\.0/perl\-5\.20\.1\.tar\.gz
>> gunzip \-c \.\./perl\-5\.20\.1\.tar\.gz | tar xvf \-
>> cd perl\-5\.20\.1/lib/unicore/
>> chmod 755 \./mktables
>> \./mktables \# Note there is no output\.\.\. just wait\. It will complete
>> sudo cp Name\.pl /usr/opt/perl5/lib/5\.20\.1/unicore
>>
>> The simple solution to this bug is to have the file packaged along with
>> Perl on AIX 7\.2\.
> A normal perl build and install will generate and install name\.pl in the
> correct place\.
>
> Your bug report doesn't make it clear whether you personally built and
> installed this perl from the tarball and something went wrong\, or whether
> you are using a pre\-packaged perl install and the file is missing from
> that packaging\.
>
> If the latter\, then you should take the issue up with whoever created the
> package\.
>
> If the former\, then you'll need to reproduce the issue with a perl
> distribution which is still maintained; preferably 5\.26\.0\.
>
> If you can show a run of  perl\-5\.26\.0\.tar\.gz being unpacked\, \./Configure
> being run with \-Dprefix='/some/dir' where /some/dir is initially empty\,
> and where name\.pl fails to be installed\, then we will look at the issue
> further\.
>
>

**Comment by p5pRT** [comment]
@iabyn - Status changed from 'open' to 'rejected'
